GEOSwiftMapboxGL/MapboxGLAnnotations.swift

```diff
@@ -17,112 +24,93 @@ public protocol GEOSwiftMapboxGL {
     func mapboxShape() -> MGLShape
 }
 
-extension Geometry : GEOSwiftMapboxGL {
-    public func mapboxShape() -> MGLShape {
-        
-        switch self {
-            
-        case is Waypoint:
-            let pointAnno = MGLPointAnnotation()
-            pointAnno.coordinate = CLLocationCoordinate2DFromCoordinate((self as! Waypoint).coordinate)
-            return pointAnno
-            
-        case is LineString:
-            var coordinates = (self as! LineString).points.map({ (point: Coordinate) ->
-                CLLocationCoordinate2D in
-                return CLLocationCoordinate2DFromCoordinate(point)
-            })
-            let polyline = MGLPolyline(coordinates: &coordinates,
-                count: UInt(coordinates.count))
-            return polyline
-            
-        case is Polygon:
-            var exteriorRingCoordinates = (self as! Polygon).exteriorRing.points.map({ (point: Coordinate) ->
-                CLLocationCoordinate2D in
-                return CLLocationCoordinate2DFromCoordinate(point)
-            })
-            
-            // interior rings are not handled by MapBoxGL, we must drop this info!
-//            let interiorRings = (self as! Polygon).interiorRings.map({ (linearRing: LinearRing) ->
-//                MKPolygon in
-//                return MKPolygonWithCoordinatesSequence(linearRing.points)
-//            })
-            
-            let polygon = MGLPolygon(coordinates: &exteriorRingCoordinates, count: UInt(exteriorRingCoordinates.count) /*, interiorPolygons: interiorRings*/)
-            return polygon
-                                     
-        case is MultiPolygon<Polygon>:
-          let mglPolygons = (self as! MultiPolygon).geometries.map({ (polygon: Polygon) -> MGLPolygon in
-            return polygon.mapboxShape() as! MGLPolygon
-          })
-          return MGLMultiPolygon(polygons: mglPolygons) 
-                                     
-        default:
-            let geometryCollectionOverlay = MGLShapesCollection(geometryCollection: (self as! GeometryCollection))
-            return geometryCollectionOverlay
-        }
-    }
+extension MGLPointAnnotation {
+  convenience init(_ coord: CLLocationCoordinate2D) {
```

Comment:
I'd probably shy away from adding a `MGLPointAnnotation` constructor that takes a `CLLocationCoordinate2D` since that's likely the type of thing that consumers would already be getting somewhere else. If you want to leave it here as an internal helper, that's fine, but let's not expose it publicly.